### ISSUE TITLE
Have an option that makes the task execute only once and cache the result for future runs.

### ISSUE THREAD

**Issue by nouex** [reecehudson/taskrunner#2]
As of now when you add a task as a dependency of another task or sequence, it is executed every time the task or sequence is run/executed. e.g.

```
var withdrawMoney = function () {
 // money withdrawal logistics
return // withdrawalReceipt;
};
seq.addTask(withdrawMoney);

// `withdrawMoney()` is executed once
seq.run()

// for some reason run again...
// `withdrawMoney()` executed **twice**
```

This could obviously be worked around by other externalities such as:

```
var withdrawalInfo;
var sMWrapper = function () {withdrawalInfo = withdrawMoney()};

seq.addTask(sMWrapper, {once: true});
// use `withdrawalInfo` alone
```

The easiest thing for the user would be to include an option that if set to true, would disactivate the function after it is executed once and use the returned value in it's place `actual`.  So all the `meta` info (index, once, ...) would stay the same except `actual`.

Note:  this is not the same as `options:once=true` since that would eliminate the dep as soon as it is used once, here we still want to use the infor returned by `withdrawMoney()` but we do not want it to execute more than once.
